homeassistant/components/switch/tahoma.py

```diff
@@ -48,4 +92,33 @@ def toggle(self, **kwargs):
     @property
     def is_on(self):
         """Get whether the switch is in on state."""
-        return False
+        if self.tahoma_device.type == 'rts:GarageDoor4TRTSComponent':
+            return False
+        return bool(self._state == STATE_ON)
+
+    @property
+    def state_attributes(self):
+        """Return the state attributes of the switch."""
+        attr = {}
+        super_attr = super().state_attributes
+        if super_attr is not None:
+            attr.update(super_attr)
+
+        if 'core:RSSILevelState' in self.tahoma_device.active_states:
+            attr[ATTR_RSSI_LEVEL] = \
+                self.tahoma_device.active_states['core:RSSILevelState']
+        if 'core:StatusState' in self.tahoma_device.active_states:
+            attr[ATTR_STATUS] = \
+                self.tahoma_device.active_states['core:StatusState']
+        if 'core:PriorityLockTimerState' in self.tahoma_device.active_states:
+            attr[ATTR_LOCK_TIMER] = \
+                self.tahoma_device.active_states['core:PriorityLockTimerState']
```

Comment:
Is this the same info as the cover?

Reply:
Yes, it is the same type of info, but from the switch. Actually I do not know, for what the PriorityLockTimerState is used in case of a switch. For example a cover is blocked by a wind sensor for a certain time, in case of strong wind.